### PR TITLE
Implemented Avro field replacer converter.

### PR DESCRIPTION
**Pull request by ibuenros**
This implements a converter that can replace string fields in Avro records for other string, supporting custom replacements. Also implemented two default replacements: constant and sha1.

### CODE REVIEW

**Comment by ibuenros** [comment]
@pcadabam can you review?

**Comment by chavdar** [comment]
I'd like to see a more detailed description of the design. 

- How do we make this dataset config driven?
- Do we allow "." notation of fields (i.e. traverse avro hierarchy)
- How do we handle different field types?

**Comment by shirshanka** [comment]
We seem to have accumulated quite a few different converters for doing field level operations on records. Instead of chaining these converters by hand and specifying config independently for each, maybe we should have an uber converter for supporting config-driven transformation of the record.

File: gobblin-core/src/main/java/gobblin/converter/avro/replacer/AvroFieldReplacer.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
```
Comment (abti) [inline gobblin-core/src/main/java/gobblin/converter/avro/replacer/AvroFieldReplacer.java]: Please change to use the Apache 2.0 License header without copyright

**Comment by abti** [comment]
https://issues.apache.org/jira/browse/GOBBLIN-74

Please update your PR title with following prefix: [GOBBLIN-74]

**Pull request closed without merging**